src/pages/dar_application/DataUseAgreements.jsx

```diff
@@ -14,11 +13,77 @@ export default function DataUseAgreements(props) {
     darCode,
     isAttested,
     cancelAttest,
+    datasets
   } = props;
+  const [daas, setDaas] = useState([]);
+
+  useEffect(() => {
+    const init = async () => {
+      try {
+        const daaList = await DAA.getDaas();
+        setDaas(daaList);
+      } catch (error) {
+        Notifications.showError({
+          text: 'Error: Unable to retrieve DAAs from server',
+        });
+      }
+    };
+    init();
+  }, []);
+
+  const RequiredDAAs = () => {
+    const fileNames = new Set();
+    const daaDivs = datasets.map((dataset) => {
+      const datasetDacId = dataset.dacId;
+      if (!datasetDacId) {
+        return <div key={dataset.id}></div>;
+      }
+      const daa = daas.find((daa) => daa.dacs.some((d) => d.dacId === datasetDacId));
+      const id = daa.daaId;
+      const fileName = daa.file.fileName.split('.')[0];
+      if (fileNames.has(fileName)) {
+        return <div key={id-dataset.name}></div>;
+      }
+      fileNames.add(fileName);
+      return (
+        <div key={id}>
+          {DAADownload(id, fileName)}
+        </div>
+      );
+    });
+    if (fileNames.size === 0) {
+      return (
+        <div></div>
+      );
+    } else {
+      return (
+        <div>
+          <h3>By submitting this data access request and in accordance with your Institution’s issuance of Library Cards to you for the agreement(s) below,</h3>
```

Comment:
Grammar nit: I think this should end in a period
```suggestion
          <h3>By submitting this data access request and in accordance with your Institution’s issuance of Library Cards to you for the agreement(s) below.</h3>
```